pkg/api/plugin_resource.go

```diff
@@ -198,6 +199,9 @@ func (hs *HTTPServer) flushStream(stream callResourceClientResponseStream, w htt
 
 		if _, err := w.Write(resp.Body); err != nil {
 			hs.log.Error("Failed to write resource response", "err", err)
+		} else if ufn != nil {
```

Comment:
This seemed like the best place to do this, as opposed to having the function return data.